### ISSUE TITLE
Sensor contact rules vs collapsing buildings or clearing trees

### ISSUE THREAD

**Issue by SuperStucco** [MegaMek/megamek#4469]
For some time now, up to 49.12, fire against buildings and trees has been limited to visual range only.  If another unit is within visual range, and the shooter is outside of visual range but within sensor range, the shooter cannot do damage to collapse a building or clear trees.  For example:

![Sensor vs building 1](https://github.com/MegaMek/megamek/assets/70451747/eb43af8a-1e59-4fcc-84bf-a8d1622f70c2)

![Sensor vs building 2](https://github.com/MegaMek/megamek/assets/70451747/1eea7639-bfb0-45f2-86e0-51a00a3030c9)

The OpFor SRM6 turret is in the same hex as a friendly artillery turret, so it has been visually spotted.  The Ryoken (Stormcrow) is within sensor range (0-10) but outside of visual range (0-5), but is not allowed to shoot at the building.  The LRMs are being fired indirectly.

It should be possible to at least shoot at turrets on buildings that are within sensor range and visually spotted by a friendly unit.

49.12 savegame:
[Sensors_vs_buildings.sav.gz](https://github.com/MegaMek/megamek/files/11593613/Sensors_vs_buildings.sav.gz)
Custom units are present (custom BA is being deployed on a later turn):
[Infantry Pack - BA.zip](https://github.com/MegaMek/megamek/files/11593618/Infantry.Pack.-.BA.zip)
[Black Knight BL-6bX1-KNT.mtf.txt](https://github.com/MegaMek/megamek/files/11593620/Black.Knight.BL-6bX1-KNT.mtf.txt)

**Comment by gsparks3** [comment]
Each individual unit that is attacking must have its target within visual range, with the possible exception of indirect fire or a C3 network spotter for a unit in the same network but outside of visual range. Therefore, this situation is correctly following the rules, although the text could possibly be changed to "outside of visual range" for clarity, as being inside or outside of sensor range has no effect on whether you can make a direct attack. 

Source: https://bg.battletech.com/forums/tactical-operations/(answered)-night-fighting-and-shooting

**Comment by SuperStucco** [comment]
No, that's not entirely right.  I can still attack mobile targets that are within sensor range but outside of visual range.  This should extend to other targets, such as buildings and especially buildings with gun emplacements.

**Comment by gsparks3** [comment]
Then the ability to attack mobile targets outside of visual range is what is wrong.
